iocore/cache/Cache.cc

```diff
@@ -2139,10 +2141,21 @@ CacheVC::is_pread_capable()
 static void
 unmarshal_helper(Doc *doc, Ptr<IOBufferData> &buf, int &okay)
 {
+  using UnmarshalFunc           = int(char *buf, int len, RefCountObj *block_ref);
+  UnmarshalFunc *unmarshal_func = &HTTPInfo::unmarshal;
+  ts::VersionNumber version(doc->v_major, doc->v_minor);
+
+  if (version < CACHE_DB_VERSION) {
+    ink_assert(CACHE_DB_VERSION_4847 == version);
+    unmarshal_func = &HTTPInfo::unmarshal_v24_1;
+  } else {
+    ink_assert(version == CACHE_DB_VERSION);
```

Comment:
I don't think `assert` is good here. The cache should keep on running even in this case. In fact, there's a check specifically for this case elsewhere - I think "Cache.cc:2298" or so.